packages/extension/src/ui/action/views/network-assets/components/custom-evm-token.vue

```diff
@@ -0,0 +1,481 @@
+          </p>
+        </div>
+      </div>
+      <div v-else-if="isValidAddress && notTokenAddress">
+        <div class="add-custom-token__error">
+          <alert-icon />
+          <p>
+            There is no ERC20 token deployed at this address. Make sure you have
+            the right contract address and you are connected to the correct
+            network.
+          </p>
+        </div>
+      </div>
+
+      <div class="add-custom-token__buttons">
+        <div class="add-custom-token__buttons-send">
+          <base-button
+            title="Add token"
+            :disabled="!isValidAddress || !tokenInfo"
+            :click="addToken"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import API from "@/providers/ethereum/libs/api";
+import { BaseNetwork } from "@/types/base-network";
+import { computed, ref, toRaw, watch } from "vue";
+import { isAddress } from "web3-utils";
+import { Erc20Token } from "@/providers/ethereum/types/erc20-token";
+import { fromBase } from "@/libs/utils/units";
+import CloseIcon from "@/ui/action/icons/common/close-icon.vue";
+import BaseButton from "@action/components/base-button/index.vue";
+import AlertIcon from "@action/icons/send/alert-icon.vue";
+import WarnIcon from "@action/icons/send/warning-icon.vue";
+import { TokensState } from "@/libs/tokens-state";
+import { CustomErc20Token, TokenType } from "@/libs/tokens-state/types";
+import { AssetsType } from "@/types/provider";
+import { formatFloatingPointValue } from "@/libs/utils/number-formatter";
+
+interface IProps {
+  network: BaseNetwork;
+  address: string;
+}
+
+const emits = defineEmits<{
+  (e: "update:close"): void;
+  (e: "update:token-added", asset: AssetsType): void;
+}>();
+
+const props = defineProps<IProps>();
+
+const tokensState = new TokensState();
+
+const contractAddress = ref<string>();
+const tokenInfo = ref<CustomErc20Token>();
+const accountBalance = ref<string>();
+const notTokenAddress = ref(false);
+const isFocus = ref(false);
+
+const isValidAddress = computed(() => {
+  if (contractAddress.value) {
+    return isAddress(contractAddress.value);
+  }
+
+  return false;
+});
+
+watch([contractAddress, props], async () => {
+  notTokenAddress.value = false;
+  tokenInfo.value = undefined;
+  accountBalance.value = undefined;
+
+  if (!props.network.customTokens) {
+    close();
+  }
+
+  if (isValidAddress.value) {
+    const api = (await props.network.api()) as API;
+
+    const info = await api.getTokenInfo(contractAddress.value!);
+
+    if (info.name !== "Unknown") {
+      let icon = props.network.icon;
+
+      try {
+        const controller = new AbortController();
+
+        const timeoutId = setTimeout(() => controller.abort(), 15_000);
+
+        const res = await fetch(
+          `https://api.coingecko.com/api/v3/coins/${props.network.coingeckoID}/contract/${contractAddress.value}`,
```

Comment:
lets use this
https://github.com/enkryptcom/enKrypt/blob/main/packages/extension/src/libs/market-data/index.ts#L58